### DISCUSSION TITLE
batch resize images, rename them and put them into a new folder?

### DISCUSSION THREAD

**Discussion by JwanKhalaf** [ImageMagick/ImageMagick#2584] [Help] [answered]
Hi all 👋 

I have something like 200 images of artists in a folder. They are all 300x300 pixels. Here is an example of how they are named `bob-dylan.jpg` and `bb-king.jpg`.

I would like to batch resize them (to 60x60 pixels), rename them and place them into another folder.

So they end up being called `bob-dylan-60.jpg` and `bb-king-60.jpg` and be moved to `/60/bob-dylan-60.jpg` ...etc

I hope that is clear. How can I achieve this with ImageMagick?

Thank you

**Comment by snibgo** [comment]
What version of IM, on what platform?
```
magick mogrify -path /60 -resize 60x60 *.jpg
```
That does most of what you want, but not the rename. So I would do it in a shell loop, calling "magick" once per input.

**Reply by JwanKhalaf** [reply]
Thank you for the reply @snibgo. I am on Windows 10 Pro (64 bit) and the version is this one: `ImageMagick-7.0.10-29-Q16-HDRI-x64-dll.exe`

**Reply by JwanKhalaf** [reply]
@snibgo I just tried your command I got the following repeated for each file:

> mogrify: unable to open image '/60\bb-king.jpg': No such file or directory @ error/blob.c/OpenBlob/3537.

**Comment by snibgo** [comment]
In Windows, the directory separator is a backslash \ not a forward slash /.

Do you have a directory "\60"? IM won't create it. You must create it first.

**Reply by JwanKhalaf** [reply]
Thanks, I tried this command and it works!

`magick mogrify -path 60 -resize 60x60 *.jpg`

But it **does not** rename the files. I believe I need to use the `-set` flag, but I can't seem to get the syntax right! Would appreciate some help with the rename aspect.

I'd like the resized files to have the original filename but with `-60` appended, so I end up with `bb-king-60.jpg`

**Comment by snibgo** [comment, marked as answer]
As I said:
> That does most of what you want, but not the rename.

Like I said, I would do it with a shell loop, for example:
```
for %F in (*.jpg) do magick %F -resize 60x60 \60\%~nF_60.jpg
```
If in a BAT script, double the percents: %%
